SECURITY.md

```diff
@@ -29,3 +35,89 @@ If you are editing this document please feel encouraged to change this format to
 - #993 Add restricted permissions to RocksDB - 1.4.5 - Anonymous security review
 - #740 Add hardening compile flags and `-fPIE` - 1.4.1 - Anonymous security review
 - #300 Add restricted permissions to osqueryd logging - 1.0.4
+
+## Security Requirements Met by Functionality
+
+### Confidentiality
+
+#### No Retrieval of Sensitive User-owned Data
+
+- Certain tables disabled by default, additional tables can be disabled
+
+#### Use of HTTPS (TLS)
+
+- HTTPS (specifically the TLS v1.2 protocol) is used to encrypt all communications
+
+#### No custom cryptography (only well-known libraries)
+
+#### No network connectivity to third parties
+
+#### Does not accept incoming network connections
+
+#### Does not 'shell out' to other binaries for data collection
+
+#### Access control functions
+
+### Integrity
+
+#### Limited attack surface
+
```

Comment:
osquery does not listen on network interfaces. If it is configured to use the network it is opt-in and always a poll model.

osquery may parse user-controlled data on disk and as made available through OS APIs. This parsing usually takes places within third-party libraries that are statically linked into osquery.

osquery responds to known vulnerabilities in third-party libraries by upgrading and making a patched release available during the next release cycle.

Reply:
osquery does reach out to the network for some tables. `curl` and the cloud ones immediately come to mind. Maybe others.